component/otelcol/processor/tail_sampling/types.go

```diff
@@ -149,158 +159,259 @@ type StringAttributeCfg struct {
+type RateLimitingConfig struct {
 	// SpansPerSecond sets the limit on the maximum nuber of spans that can be processed each second.
 	SpansPerSecond int64 `river:"spans_per_second,attr"`
 }
 
-func (rateLimitingCfg RateLimitingCfg) Convert() tsp.RateLimitingCfg {
-	var otelCfg tsp.RateLimitingCfg
+func (rateLimitingConfig RateLimitingConfig) Convert() tsp.RateLimitingCfg {
+	var otelConfig tsp.RateLimitingCfg
 
 	mustDecodeMapStructure(map[string]interface{}{
-		"spans_per_second": rateLimitingCfg.SpansPerSecond,
-	}, &otelCfg)
+		"spans_per_second": rateLimitingConfig.SpansPerSecond,
+	}, &otelConfig)
 
-	return otelCfg
+	return otelConfig
 }
 
-// SpanCountCfg holds the configurable settings to create a Span Count filter sampling policy
+// SpanCountConfig holds the configurable settings to create a Span Count filter sampling policy
 // sampling policy evaluator
-type SpanCountCfg struct {
+type SpanCountConfig struct {
 	// Minimum number of spans in a Trace
 	MinSpans int32 `river:"min_spans,attr"`
 }
 
-func (spanCountCfg SpanCountCfg) Convert() tsp.SpanCountCfg {
-	var otelCfg tsp.SpanCountCfg
+func (spanCountConfig SpanCountConfig) Convert() tsp.SpanCountCfg {
+	var otelConfig tsp.SpanCountCfg
+
+	mustDecodeMapStructure(map[string]interface{}{
+		"min_spans": spanCountConfig.MinSpans,
+	}, &otelConfig)
+
+	return otelConfig
+}
+
+// BooleanAttributeConfig holds the configurable settings to create a boolean attribute filter
+// sampling policy evaluator.
+type BooleanAttributeConfig struct {
+	// Tag that the filter is going to be matching against.
+	Key string `river:"key,attr"`
+	// Value indicate the bool value, either true or false to use when matching against attribute values.
+	// BooleanAttribute Policy will apply exact value match on Value
+	Value bool `river:"value,attr"`
+}
+
+func (booleanAttributeConfig BooleanAttributeConfig) Convert() tsp.BooleanAttributeCfg {
+	var otelConfig tsp.BooleanAttributeCfg
+
+	mustDecodeMapStructure(map[string]interface{}{
+		"key":   booleanAttributeConfig.Key,
+		"value": booleanAttributeConfig.Value,
+	}, &otelConfig)
+
+	return otelConfig
+}
+
+// The error mode determines whether to ignore or propagate
+// errors with evaluating OTTL conditions.
+type ErrorMode string
+
+const (
+	// "ignore" causes evaluation to continue to the next statement.
+	ErrorModeIgnore ErrorMode = "ignore"
+	// "propagate" causes the evaluation to be false and an error is returned.
+	ErrorModePropagate ErrorMode = "propagate"
+)
+
+var (
+	_ river.Validator          = (*ErrorMode)(nil)
+	_ encoding.TextUnmarshaler = (*ErrorMode)(nil)
+)
+
+// Validate implements river.Validator.
+func (e *ErrorMode) Validate() error {
+	if e == nil {
+		return nil
+	}
+
+	var ottlError ottl.ErrorMode
+	return ottlError.UnmarshalText([]byte(string(*e)))
+}
+
+// Convert the River type to the Otel type
+func (e *ErrorMode) Convert() ottl.ErrorMode {
+	if e == nil || *e == "" {
+		return ottl.ErrorMode("")
+	}
```

Comment:
@rfratto - after your review I had to fix a linter issue, and I added this clause. It feels dodgy because `ottl.ErrorMode("")` is not a valid error mode, but I can't think of a better way to handle this. I'm just going to merge it as it is, I hope you don't mind.